### PR TITLE
feat: add new gmc response code TIS21-3583

### PR DESCRIPTION
**Pull request by jayanta2018**
- added the new gmc response code 161
- added a test to check if that message is passed through

### CODE REVIEW

**Review by CaiWillis — COMMENTED** [review]
LGTM

File: application/src/test/java/uk/nhs/hee/tis/revalidation/service/RecommendationServiceTest.java [hunk 1]
```diff
@@ -587,6 +588,22 @@ void shouldNotUpdateRecommendationWhenSubmitFail() {
...
+        UNDER_REVIEW);
+    final var userProfileDto = buildRoUserProfileDto(gmcNumber1);
+    when(doctorsForDBRepository.findById(gmcNumber1)).thenReturn(Optional.of(doctorsForDB));
+    when(recommendationRepository.findByIdAndGmcNumber(recommendationId, gmcNumber1))
+        .thenReturn(recommendation);
+    when(gmcClientService.submitToGmc(doctorsForDB, recommendation, userProfileDto))
+        .thenReturn(buildRecommendationV2Response(DOCTOR_DEFERRAL_ON_EARLY_DATE.getCode()));
+
+    assertThrows(RecommendationException.class, () -> recommendationService
+        .submitRecommendation(recommendationId, gmcNumber1, userProfileDto));
```
Comment (JosephKelly) [inline application/src/test/java/uk/nhs/hee/tis/revalidation/service/RecommendationServiceTest.java:603]: There's nothing here to actually assert the new code.
As well as here, you could consider how the Controller responds `RecommendationControllerTest.java`

**Review by JosephKelly — COMMENTED** [review]
The test checks that an exception is thrown, not the message or even that the application will respond to a client with that message.

IMO, that's doable in this PR but I'm stopping at suggesting, rather than requesting, it as a change.

**Review by JosephKelly — DISMISSED** [review]
Looking good.  Will review again if you're adding that extra type of test.  Pls don't force push once someone's looking at your PR

File: application/src/test/java/uk/nhs/hee/tis/revalidation/service/RecommendationServiceTest.java [hunk 2]
```diff
@@ -825,10 +848,13 @@ void shouldGetRecommendationStatusCheckDtos() {
     final var recommendation = buildRecommendation(gmcId, recommendationId, SUBMITTED_TO_GMC,
         UNDER_REVIEW);
 
-    when(recommendationRepository.findAllByRecommendationStatus(RecommendationStatus.SUBMITTED_TO_GMC))
+    when(recommendationRepository
```
Comment (jayanta2018) [inline application/src/test/java/uk/nhs/hee/tis/revalidation/service/RecommendationServiceTest.java:851]: The changes which are seen next are produced due to the "reformat code" button. They can be ignored.

File: application/src/main/java/uk/nhs/hee/tis/revalidation/messages/RabbitMessageListener.java [hunk 3]
```diff
@@ -60,7 +60,7 @@ public class RabbitMessageListener {
   @RabbitListener(queues = "${app.rabbit.queue}")
   public void receivedMessage(final DoctorsForDbDto gmcDoctor) {
     try {
-      log.info("DoctorsForDbDto message received from rabbit: {}", gmcDoctor);
+      log.debug("DoctorsForDbDto message received from rabbit: {}", gmcDoctor);
```
Comment (JosephKelly) [inline application/src/main/java/uk/nhs/hee/tis/revalidation/messages/RabbitMessageListener.java:63]: @CaiWillis Just an FYI as I think you'd want to know when this is in stage/prod

**Review by JosephKelly — APPROVED** [review]

**Review by CaiWillis — APPROVED** [review]

**Pull request merged**